lib/resources/collection/index.js

```diff
@@ -637,11 +649,18 @@ function createDomain(data, errors) {
   return domain;
 }
 
-Collection.prototype.doAfterCommitEvent = function(method, ctx, data){
+Collection.prototype.doAfterCommitEvent = function(method, ctx, data, previous, protectedKeys) {
   var collection = this;
   if (collection.shouldRunEvent(collection.events.AfterCommit, ctx)) {
-    collection.events.AfterCommit.run(ctx, {data: data, 'this': data, method: method}, function (err) { 
-      if (err) debug('AfterCommit errors in script: %j', err); 
+    data = _.clone(data);
+    if (protectedKeys && protectedKeys.length > 0 && previous) {
+      // add back whatever fields were protected, because they are removed from data
+      protectedKeys.forEach(function (key) {
+        data[key] = previous[key];
+      });
+    }
```

Comment:
Line 655-661 
```
    data = _.clone(data);
    if (protectedKeys && protectedKeys.length > 0 && previous) {
      // add back whatever fields were protected, because they are removed from data
      protectedKeys.forEach(function (key) {
        data[key] = previous[key];
      });
    }
```
only will be exec when called from line 563
```
collection.doAfterCommitEvent('PUT', ctx, item, prev, domain.protectedKeys);
```

Shall we move this part of code to there instead?